fission/src/systems/match_mode/RobotPositionTracker.ts

```diff
@@ -0,0 +1,64 @@
+import MirabufSceneObject from "@/mirabuf/MirabufSceneObject"
+import { MiraType } from "@/mirabuf/MirabufLoader"
+import SceneRenderer from "../scene/SceneRenderer"
+import SimulationSystem from "@/systems/simulation/SimulationSystem"
+import World from "../World"
+import { convertJoltMat44ToThreeMatrix4 } from "@/util/TypeConversions"
+import JOLT from "@/util/loading/JoltSyncLoader"
+import * as THREE from "three"
+
+class RobotPositionTracker {
+    private static _mapBoundaryY: number = -4
+    private static _offMapPenalty: number = 0
+
+    public static update(sceneRenderer: SceneRenderer): void {
+        const robots = [...sceneRenderer.sceneObjects.values()].filter(
+            (obj): obj is MirabufSceneObject => obj instanceof MirabufSceneObject && obj.miraType === MiraType.ROBOT
+        )
+
+        robots.forEach(robot => {
+            const rootNodeId = robot.getRootNodeId()
+            if (!rootNodeId) {
+                return
+            }
+
+            const rootBody = World.physicsSystem.getBody(rootNodeId)
+            const rootTransform = convertJoltMat44ToThreeMatrix4(rootBody.GetWorldTransform())
+
+            const rootPosition = new THREE.Vector3()
+            const rootRotation = new THREE.Quaternion()
+            const rootScale = new THREE.Vector3()
+            rootTransform.decompose(rootPosition, rootRotation, rootScale)
+
+            if (rootPosition.y < this._mapBoundaryY) {
+                console.log(`Robot fell off the map: ${robot.assemblyName}`)
+                SimulationSystem.robotPenalty(robot, this._offMapPenalty, "Robot fell off the map")
+
+                // TODO: Once driver station is implemented, we should reset the robot to the driver station position
+                const resetPosition = new JOLT.RVec3(0, 0.2, 0)
+                const resetRotation = JOLT.Quat.prototype.sIdentity()
+                const zeroVelocity = new JOLT.Vec3(0, 0, 0)
```

Comment:
Yeah my concern about resetting to 0,0 is that for fields like 2018 you just get stuck in something. Resetting to driver station positions would solve that once they're implemented, created `AARD-2040` so we don't forget about it